kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/data/ColumnVectorConverter.java

```diff
@@ -0,0 +1,210 @@
+        boolean[] isNull = new boolean[c.getSize()];
+        DataType d = c.getDataType();
+        for (int i = 0; i < c.getSize(); i++) {
+            if (c.isNullAt(i)) {
+                isNull[i] = true;
+            } else {
+                if (d instanceof ByteType) {
+                    values[i] = c.getByte(i);
+                } else if (d instanceof ShortType) {
+                    values[i] = c.getShort(i);
+                } else {
+                    String msg = String.format(
+                            "`%s` cannot convert to int`",
+                            d);
+                    throw new IllegalStateException(msg);
+                }
+            }
+        }
+        return new DefaultIntVector(
+                IntegerType.INTEGER, c.getSize(), Optional.of(isNull), values);
+    }
+    protected static ColumnVector convertToLong(ColumnVector c) {
+        long[] values = new long[c.getSize()];
+        boolean[] isNull = new boolean[c.getSize()];
+        DataType d = c.getDataType();
+        for (int i = 0; i < c.getSize(); i++) {
+            if (c.isNullAt(i)) {
+                isNull[i] = true;
+            } else {
+                if (d instanceof ByteType) {
+                    values[i] = c.getByte(i);
+                } else if (d instanceof ShortType) {
+                    values[i] = c.getShort(i);
+                } else if (d instanceof IntegerType) {
+                    values[i] = c.getInt(i);
+                } else if (d instanceof StringType) {
+                    values[i] =  Long.parseLong(c.getString(i));
+                } else {
+                    String msg = String.format(
+                            "`%s` cannot convert to long`",
+                            d);
+                    throw new IllegalStateException(msg);
+                }
+            }
+        }
+        return new DefaultLongVector(
+                LongType.LONG, c.getSize(), Optional.of(isNull), values);
+    }
+    protected static ColumnVector convertToDecimal(ColumnVector c) {
+        BigDecimal[] values = new BigDecimal[c.getSize()];
+        boolean[] isNull = new boolean[c.getSize()];
+        DataType d = c.getDataType();
+        for (int i = 0; i < c.getSize(); i++) {
+            if (d instanceof ByteType) {
+                values[i] = new BigDecimal(c.getByte(i));
+            } else if (d instanceof ShortType) {
+                values[i] = new BigDecimal(c.getShort(i));
+            } else if (d instanceof IntegerType) {
+                values[i] = new BigDecimal(c.getInt(i));
+            } else if (d instanceof LongType) {
+                values[i] = new BigDecimal(c.getLong(i));
+            } else {
+                String msg = String.format(
+                        "`%s` cannot convert to big decimal`",
+                        d);
+                throw new IllegalStateException(msg);
+            }
+        }
+        return new DefaultDecimalVector(
```

Comment:
There is some choice here if original column have non default `int precision, int scale` should we honor that when we do the conversion?